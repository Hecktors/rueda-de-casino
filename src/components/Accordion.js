import { useEffect, useState } from 'react'
import styled from 'styled-components/macro'
import PropTypes from 'prop-types'
import { ReactComponent as ArrowRightIcon } from '../assets/img/arrow_right.svg'
import { ReactComponent as ArrowDownIcon } from '../assets/img/arrow_down.svg'

LevelAccordion.propTypes = {
  name: PropTypes.string.isRequired,
  moves: PropTypes.array.isRequired,
  selectedMoves: PropTypes.array.isRequired,
  userInput: PropTypes.array.isRequired,
  updateUserInput: PropTypes.func.isRequired,
}

export default function LevelAccordion({
  name,
  moves,
  selectedMoves,
  userInput,
  updateUserInput,
}) {
  const selectedMoveIds = selectedMoves.map((move) => move.id)
  const hasSelectedMove = moves.some((move) =>
    selectedMoveIds.includes(move.id)
  )

  const [isOpen, setIsOpen] = useState()

  useEffect(() => {
    hasSelectedMove && setIsOpen(true)
  }, [hasSelectedMove])

  const listItems = moves.map((move) => (
    <li key={move.id}>
      <label className={userInput.includes(move.id) ? 'isChecked' : ''}>
        <input
          onChange={() => updateUserInput(move.id)}
          type="checkbox"
          checked={userInput.includes(move.id)}
        />{' '}
        {move.name}
      </label>
    </li>
  ))

  function toogleLevelList() {
    setIsOpen(!isOpen)
  }

  return (
    <LevelAccordionStyled isOpen={isOpen}>
      <h3 onClick={toogleLevelList}>
        {isOpen ? <ArrowDownIcon /> : <ArrowRightIcon />}{' '}
        <span className="level-name">{name.toUpperCase()}</span>
      </h3>
      <ul>{listItems}</ul>
    </LevelAccordionStyled>
  )
}

const LevelAccordionStyled = styled.div`
  padding: 10px;
  border-radius: 5px;
  margin-bottom: 10px;
  background-color: var(--bg-color-accordion);

  input {
    display: none;
    margin-right: 5px;
  }

  h3 {
    font-size: 1rem;
    padding: 0;
    display: flex;
    justify-content: space-between;

    & .level-name {
      flex-grow: 1;
      font-size: inherit;
      font-weight: normal;
      text-align: center;
    }

    svg {
      transform: scale(2);
    }
  }

  ul {
    overflow: hidden;
    height: ${(props) => (props.isOpen ? 'auto' : 0)};
  }

  li {
    color: var(--color-listitem);
    padding: 3px 0;

    &:first-of-type {
      padding-top: 15px;
    }

    label {
      display: inline-block;
      width: 100%;
    }
  }

  .isChecked {
    color: var(--color-listitem-active);
  }
`
