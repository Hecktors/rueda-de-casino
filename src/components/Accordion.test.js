import ReactDOM from 'react-dom'
import Accordion from './Accordion'
import { render } from '@testing-library/react'
const moves = [
  {
    id: 1,
    name: 'la prima',
    steps: 2,
    filename: 'la_prima.mp3',
  },
  {
    id: 2,
    name: 'sombrero',
    steps: 2,
    filename: 'sombrero.mp3',
  },
]

describe('Accordion', () => {
  it('renders Accordion', () => {
    const div = document.createElement('div')
    ReactDOM.render(
      <Accordion
        name="test accordion"
        moves={moves}
        selectedMoves={[]}
        userInput={[2]}
        updateUserInput={() => {}}
      />,
      div
    )
    expect(div.firstChild).toMatchSnapshot()
  })
})
