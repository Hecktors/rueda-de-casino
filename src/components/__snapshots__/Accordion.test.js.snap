// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion renders Accordion 1`] = `
<div
  class="Accordion__LevelAccordionStyled-sn9tuv-0 khvdae"
>
  <h3>
    <svg>
      arrow_right.svg
    </svg>
     
    <span
      class="level-name"
    >
      TEST ACCORDION
    </span>
  </h3>
  <ul>
    <li>
      <label
        class=""
      >
        <input
          type="checkbox"
        />
         
        la prima
      </label>
    </li>
    <li>
      <label
        class="isChecked"
      >
        <input
          checked=""
          type="checkbox"
        />
         
        sombrero
      </label>
    </li>
  </ul>
</div>
`;
